### ISSUE TITLE
Integration test should run all steps to verify implementation

### ISSUE THREAD

**Issue by iamogbz** [tophat/semantic-release-firefox-add-on#95]
Configure the e2e folder to build and upload the fake add-on

1. Set up the CI of the e2e to run semantic-release with `dryRun: false` and actually publish a fake add-on.
    - It should be avoided to publish a fake add-on to AMO. [`addons-dev.allizom.org`](https://addons-dev.allizom.org) seems available instead ([ref](https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#working-on-web-ext-sign)).

_Originally posted by @iorate in https://github.com/tophat/semantic-release-firefox-add-on/pull/94#issuecomment-601141828_

**Comment by iorate** [comment]
> Although moving it into another repo wouldn't quite work as we want to run the tests before merging or releasing.

I wonder if it is possible to run the e2e test at any time by moving it to another repo and cloning it as a subdirectory.

File hierarchy:
```
semantic-release-firefox-add-on
+-- .git
+-- .gitignore
+-- e2e
| +-- .git
| +-- manifest.json
| `-- package.json
+-- node_modules
`-- src
```

/.gitignore
```
/e2e/
```

/e2e/package.json
```json
{
  "scripts": {
    "test-e2e": "git commit --allow-empty -m 'feat: dummy' && FIREFOX_API_KEY='dummy' FIREFOX_SECRET_KEY='dummy' ../node_modules/.bin/semantic-release --no-ci",
    "comment": "should we pull before test and push after test?"
  },
  "release": {
    "plugins": [
      [
        "../src/index.js",
	{
	  "...options...": "...values...",
          "someOptionToSetServer": "addons-dev.allizom.org"
	}
      ]
    ]
  }
}
```

To run the e2e test with the working tree:
```shell
cd e2e && yarn test-e2e
```

**Comment by iamogbz** [comment]
Should be resolved by 00ef7540496570283d890c5ec2a55a0410183209
